crates/hyperqueue/src/dashboard/ui/screens/worker/worker_tasks_table.rs

```diff
@@ -0,0 +1,110 @@
+use crate::common::format::human_duration;
+use crate::dashboard::data::tasks_timeline::TaskInfo;
+use crate::dashboard::ui::terminal::DashboardFrame;
+use crate::dashboard::ui::widgets::table::{StatefulTable, TableColumnHeaders};
+use chrono::{DateTime, Local};
+use std::time::SystemTime;
+use tako::TaskId;
+use tui::layout::{Constraint, Rect};
+use tui::widgets::{Cell, Row};
+
+#[derive(Default)]
+pub struct WorkerTasksTable {
+    table: StatefulTable<WorkerTaskRow>,
+}
+
+impl WorkerTasksTable {
+    pub fn update(&mut self, tasks_info: Vec<&TaskInfo>) {
+        let mut rows = create_rows(tasks_info);
+        rows.sort_by_key(|task_row| task_row.task_state.len());
```

Comment:
Why are you sorting based on the task state length? :thinking: We should definitely store the task state in the task timeline to distinguish between waiting/running/finished/failed (even though waiting tasks won't be shown in this table).